### PR TITLE
[Feature] use return_mask as a string in pad_sequence  

### LINKED ISSUE #726: [Feature Request] Using `pad_sequence(... return_mask=True)` leads to confusing debug sessions
## Motivation

Working on computer vision applications, I often encounter situations where certain object models contain a key called "masks", most commonly referring to something like semantic segmentation outputs.

The function `pad_sequence` has an argument `return_mask` which adds a key called also called "masks". This can lead to some confusing behavior or bugs, especially since this may go unnoticed and could be hard to trace back. 

Currently, I cannot control the key that the padding mask is being written to. In general, there is no clear way to separate/modify keys that TensorDict internally uses and my own user keys.

## Solution

Make the interface with `return_mask=True` return the valid-entry mask as a secondary output, e.g.
```python
def pad_sequence(..., return_mask=False) -> TensorDict: ...  # padded tensordict
def pad_sequence(..., return_mask=True) -> tuple[TensorDict, TensorDict]: ... # padded tensordict, valid masks
```

## Alternatives

1. Propose to have all internally used keys exposed as constants and prepended with an underscore, e.g. export a variable `KEY_MASK: Final = "_mask"`, to clearly differentiate internal keys from user keys. 
2. Propose to pass the key (with default "masks") as an argument, either by a value of `str` to `return_mask` or as a separate parameter.
3. Copy-paste the method to my own codebase and change the key.

## Additional context

I understand that this is a minor/niche request, but since PyTorch is central to many computer vision applications, I hope that the adoption of TensorDict could be helped by avoiding the most common keys that users would be inclined to use.

## Checklist

- [x ] I have checked that there is no similar issue in the repo (**required**)

### PR DESCRIPTION
**Pull request by dtsaras**
## Description

This PR augments the ```pad_sequence``` function by allowing the parameter ```return _mask``` to be either a ```bool ``` or ```str```. In case ```return_mask``` is a ```bool```, the function behaves as before, but if ```return_mask``` is a ```str```, then the provided string becomes the key for the padding masks instead of "masks".

Additionally, for debugging purposes, if the TensorDict contains scalar tensors i.e. ```torch.Size([])```, the function raises a runtime error that the method does not support padding scalar tensors. This is consistent with ```torch.nn.utils.rnn.pad_sequence```. In case, we would like to support this we could just unqueeze such tensors. 

## Motivation and Context

In case the user already has some masks within the TensorDict under a key "masks" this will prevent the issue of renaming the masks such that they won't be overwritten by this method.

`close #726`

- [x] I have raised an issue to propose this change ([required](https://github.com/pytorch/tensordict/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Remove all that do not apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)
- [ ] Example (update in the folder of examples)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/pytorch/tensordict/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: tensordict/functional.py [hunk 1]
```diff
@@ -97,7 +97,7 @@ def pad_sequence(
             written.
         device (device compatible type, optional): if provded, the device where the
             TensorDict output will be created.
-        return_mask (bool, optional): if ``True``, a "masks" entry will be returned.
+        return_mask (bool, str): if ``True``, a "masks" entry will be returned. If ``return_mask`` is a string, it will be return the masks and be used as the key for the masks entry.
```
Comment (vmoens) [inline tensordict/functional.py:100]: could be a nested key

File: tensordict/functional.py [hunk 2]
```diff
@@ -132,6 +132,11 @@ def pad_sequence(
     if not list_of_tensordicts:
         raise RuntimeError("list_of_tensordicts cannot be empty")
 
+    masks_key = "masks"
+    if isinstance(return_mask, str):
+        masks_key = return_mask
+        return_mask = True
```
Comment (vmoens) [inline tensordict/functional.py:138]:
```suggestion
    masks_key = "masks"
    if not isinstance(return_mask, bool):
        masks_key = unravel_key(return_mask)
        return_mask = True
```

File: test/test_tensordict.py [hunk 3]
```diff
@@ -1085,7 +1085,7 @@ def test_pad(self):
         assert torch.equal(padded_td["a"], expected_a)
         padded_td._check_batch_size()
 
-    @pytest.mark.parametrize("make_mask", [True, False])
+    @pytest.mark.parametrize("make_mask", [True, "bibbidi-bobbidi-boo", False])
```
Comment (vmoens) [inline test/test_tensordict.py:1088]:
```suggestion
    @pytest.mark.parametrize("make_mask", [True, ("bibbidi", "bobbidi", "boo"), False])
```

**Review by vmoens — APPROVED** [review]
LGTM let's just generalize it to nested keys

**Comment by vmoens** [comment]
> @vmoens I think it's good to merge, thanks for the suggestions. One thing I noticed is that there is a device parameter for the function that does not get used anywhere in the code. We should fix that maybe in a separate MR.

Definitely!

File: tensordict/functional.py [hunk 4]
```diff
@@ -84,7 +84,7 @@ def pad_sequence(
     padding_value: float = 0.0,
     out: T | None = None,
     device: DeviceType | None = None,
-    return_mask: bool | None = False,
+    return_mask: bool | str = False,
```
Comment (vmoens) [inline tensordict/functional.py:87]:
```suggestion
    return_mask: bool | NestedKey = False,
```

File: tensordict/functional.py (hunk 1 above)
Comment (vmoens) [inline tensordict/functional.py:100]:
```suggestion
        return_mask (bool or NestedKey, optional): if ``True``, a "masks" entry will be returned. If ``return_mask`` is a nested key (string or tuple of strings), it will be return the masks and be used as the key for the masks entry.
```

**Review by vmoens — APPROVED** [review]
